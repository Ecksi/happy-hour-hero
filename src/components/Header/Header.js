import React, { Component } from 'react';
import { connect } from 'react-redux';
import './Header.css';
import SearchBar from '../SearchBar/SearchBar';

class Header extends Component {
  constructor (props) {
    super(props);

    this.state = {
      dropdownSelected: false
    }
    
  }

  handleSubmit = () => {
    this.setState({
      dropdownSelected: true
    });
  }

  render() {
    const { address } = this.props.location;

    return (
      <header>
        <section className="headerContainer">
          <i className="fas fa-bars"></i>
          <div className="headerLocation">
            <p>{ address }</p>
            <i className="fas fa-caret-down" onClick={ this.handleSubmit }></i>
            { this.state.dropdownSelected ? <SearchBar /> : null }
          </div>
        </section>
      </header>
    );
  }
}

export const mapStateToProps = (state) => ({
  location: state.location
});

export default connect(mapStateToProps)(Header);
