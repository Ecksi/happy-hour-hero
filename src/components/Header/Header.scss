header {
  height: 75px;
  display: flex;
  background: #fdb82a;
  align-items: center;
}

.headerContainer {
  display: flex;
  width: 90%;
  margin: 0 auto;
  align-items: center;
}

.headerLocation {
  margin-left: 50px;
  display: flex;
  align-items: center;
}

.fa-bars {
  color: #fff;
  font-size: 1.5rem;
}

.headerLocation p {
  color: #fff;
  font-size: 1.5rem;
}

.headerLocation i {
  color: #fff;
  font-size: 1.5rem;
  margin-left: 20px;
}
