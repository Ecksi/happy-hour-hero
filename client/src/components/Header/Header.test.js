import React from 'react';
import { shallow } from 'enzyme';
import { Header } from './Header';

describe('Header', () => {
  let wrapper;
  let mockProps;

  beforeEach(() => {
    mockProps = {
      location: {address: 'Turing School Denver, CO'},
    };
    wrapper = shallow(<Header {...mockProps} />);
  });

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleSubmit', () => {
    it('should set of dropdownSelected to true', () => {
      wrapper.instance().handleSubmit();

      expect(wrapper.state('dropdownSelected')).toEqual(true);
    });
  });

  describe('handleSelected', () => {
    it('should set of dropdownSelected to false', () => {
      wrapper.instance().handleSelected();

      expect(wrapper.state('dropdownSelected')).toEqual(false);
    });
  });
});