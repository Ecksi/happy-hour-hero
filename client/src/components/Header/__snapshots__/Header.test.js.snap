// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header
    location={
        Object {
            "address": "some where",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <section
        className="headerContainer"
>
        <i
                className="fas fa-bars"
        />
        <div
                className="headerLocation"
        >
                <p>
                        some where
                </p>
                <i
                        className="fas fa-caret-down"
                        onClick={[Function]}
                />
        </div>
</section>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="fas fa-bars"
/>,
          <div
            className="headerLocation"
>
            <p>
                        some where
            </p>
            <i
                        className="fas fa-caret-down"
                        onClick={[Function]}
            />
</div>,
        ],
        "className": "headerContainer",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fas fa-bars",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <p>
                some where
</p>,
              <i
                className="fas fa-caret-down"
                onClick={[Function]}
/>,
              null,
            ],
            "className": "headerLocation",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "some where",
              },
              "ref": null,
              "rendered": "some where",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-caret-down",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            null,
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <section
          className="headerContainer"
>
          <i
                    className="fas fa-bars"
          />
          <div
                    className="headerLocation"
          >
                    <p>
                              some where
                    </p>
                    <i
                              className="fas fa-caret-down"
                              onClick={[Function]}
                    />
          </div>
</section>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <i
              className="fas fa-bars"
/>,
            <div
              className="headerLocation"
>
              <p>
                            some where
              </p>
              <i
                            className="fas fa-caret-down"
                            onClick={[Function]}
              />
</div>,
          ],
          "className": "headerContainer",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "fas fa-bars",
            },
            "ref": null,
            "rendered": null,
            "type": "i",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <p>
                  some where
</p>,
                <i
                  className="fas fa-caret-down"
                  onClick={[Function]}
/>,
                null,
              ],
              "className": "headerLocation",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "some where",
                },
                "ref": null,
                "rendered": "some where",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-caret-down",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              null,
            ],
            "type": "div",
          },
        ],
        "type": "section",
      },
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
